src/decisionengine/framework/engine/SourceWorkers.py

```diff
@@ -174,29 +174,45 @@ def update(self, channel_name, source_configs):
                 worker = SourceWorker(key, config, channel_name, self._exchange, self._broker_url)
                 self._workers[key] = worker
                 workers[key] = worker
+                self._use_count[key] = {channel_name}
 
         return workers
 
-    def prune(self, source_names):
+    def unique(self, source_name):
+        with self._lock:
+            return len(self._use_count[source_name]) == 1
+
+    def detach_channel(self, channel_name, source_names):
```

Comment:
just `detach()` to be consistent w/ prune where channel is not mentioned

Reply:
Will change